backend/game/player/player.py

```diff
@@ -6,14 +7,21 @@
 
 
 class Player:
-    def __init__(self):
+    def __init__(self, username=None):
```

Comment:
Something you can do here is set the default to "guest_{}".format(random.randint(1000, 9999)) that way you can avoid the if statement, since you know what the default is and its never going to be None

Reply:
That's what I tried at first. The problem is that random number is generated one time only so all players had the same random number for every test. The random number generation had to be moved inside the class so it would wait until object creation to actually get a random number. 

my first attempt was `def __init__(self, username="guest_{}".format(random.randint(1000, 9999))` but all the tests broke